pages: update `nextjs` framework guide (#2093)
cloudflare/cloudflare-docs: products/pages/src/content/framework-guides/deploy-a-nextjs-site.md

## Creating a new project

Create a new project using `npx` or `yarn` by running either of the following commands in your terminal:

```sh
$ npx create-next-app --example with-static-export my-app
# or
$ yarn create next-app --example with-static-export my-app
```

> Once complete, a new `my-app` directory will be created using the official [`with-static-export`](https://github.com/vercel/next.js/tree/canary/examples/with-static-export) example as a template.

Reviewer: ```suggestion
After creating your project, a new `my-app` directory will be generated using the official [`with-static-export`](https://github.com/vercel/next.js/tree/canary/examples/with-static-export) example as a template.
```

| Configuration option | Value           |
| -------------------- | ----------------|
| Production branch    | `main`           |

> | Build command        | `npm run export` |

Reviewer 2: if we start using the `with-static-export` template, is there a way we can make this table work for both new (using that template) and old (bringing their own code) users? btw, this also may introduce a change to our UI via the framework `<select>` when a user is creating a project - cc @1000hz

Author: I don't know what the current next preset is off hand, but it should always have `next export` involved. 

A recent version changed jt from `next export` alone to `next build && next export`, which this template does via npm scripts.

I think using this template is safer in the long run since they'll be more inclined to keep this working as is, whereas the default template will always be changing. In order to get it to work now, you need to hard code a bunch of paths into the config file which is (a) messy and (b) not sure if it's responsibility to teach people how to use Next. These are quick starts.

Not sure how this interfaces with the dashboard or how it affects existing users; I'm sure I'm missing something

Reviewer 2: > Not sure how this interfaces with the dashboard or how it affects existing users; I'm sure I'm missing something

ignore me! i went and double checked, the framework preset when creating a new pages project does `next build && next export`, so it's not coupled to this in any way. i think those instructions were derived from this guide at one point, but we should be ok to update this

Author: Ah gotcha. I'm really only updating this because our current guide is broken as of next@11. @nevikashah discovered this when trying to follow it & I was able to reproduce the same
